src/axom/core/ArrayBase.hpp

```diff
@@ -408,7 +446,7 @@ class ArrayBase
     const IndexType baseIdx =
       numerics::dot_product((const IndexType*)idx, m_strides.begin(), DIM);
     assert(inBounds(baseIdx));
-    return asDerived().data()[baseIdx];
+    return asDerived().data()[baseIdx * m_spacing];
```

Comment:
In the multi-dimensional case, I think if you set `m_strides[DIM - 1] = m_spacing` in the constructor and `updateStrides()`, you can avoid the additional multiplication by `m_spacing` here.

Reply:
That's a great idea, and that's how I first coded it.  Unfortunately, there was a problem with the assert on line 448.  With `m_spacing` multiplied into the `baseIdx`, it was no longer "in bounds".  I decided to clarify that that by "strides", we mean the logical stride, which is the same regardless of the spacing value.  We'd just multiply by the spacing when computing the memory offset, as in line 449.